manim/utils/hashing.py

```diff
@@ -324,7 +328,7 @@ def get_json(obj: dict):
 def get_hash_from_play_call(
     scene_object: Scene,
     camera_object: Camera | OpenGLCamera,
-    animations_list: Iterable[Animation],
+    animations_list: Iterable[Animation] | None,
```

Comment:
An issue with this is that `animations_list` is expected to be an iterable inside the function. If it's `None`, it will crash.

I noticed that `CairoRenderer.play(self, scene, *args, **kwargs)` passes `scene.animations` which is typed as `list[Animation] | None`. However, since `scene.compile_animation_data(*args, **kwargs)` is called before that, `scene.animations` will always be a `list[Animation]` at that point.

Therefore, instead of typing `animations_list` as `Iterable[Animation] | None`, you have to make an assertion inside `CairoRenderer.play()` that `scene.animations is not None` before the call to `get_hash_from_play_call()`.